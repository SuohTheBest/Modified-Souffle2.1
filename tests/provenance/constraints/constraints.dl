// Souffle - A Datalog Compiler
// Copyright (c) 2017, The Souffle Developers. All rights reserved
// Licensed under the Universal Permissive License v 1.0 as shown at:
// - https://opensource.org/licenses/UPL
// - <souffle root>/licenses/SOUFFLE-UPL.txt

// This code tests the provenance explain interface for a simple example containing constraints.

.pragma "provenance" "explain"

.decl Rule(n: number)
Rule(0).

.decl neg(x:number)
neg(121).

.decl Patient(SSN: number, ID: number)
.input Patient

.decl Violation(SSN: number, ID1: number, ID2: number)
.output Violation

Violation(ssn, p1, p2) :- Patient(ssn, p1), !neg(ssn), Patient(ssn, p2), p1 != p2, Rule(0).
