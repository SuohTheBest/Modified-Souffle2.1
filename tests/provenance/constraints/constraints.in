explain Violation(123,1,2)
explainnegation Violation(123,1,1)
1
