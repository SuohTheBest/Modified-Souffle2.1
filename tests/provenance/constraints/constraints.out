Rule(0) !neg(123) Patient(123, 1) Patient(123, 2) 1 != 2 
-----------------------------------------------------(R1)
                  Violation(123, 1, 2)                   
Rule(0) ✓ !neg() ✓ Patient(123, 1) ✓ Patient(123, 1) ✓ 1 != 1 x 
--------------------------------------------------------------------(R1)
                           Violation(123,1,1)                           
